packages/dev/src/commands/setup.js

```diff
@@ -0,0 +1,199 @@
+ * @param {import('../utils/getConfiguration.js').Configuration} configuration
+ * @returns {Promise<import('../utils/getConfiguration.js').InstanceConfiguration>}
+ */
+async function getInstanceConfiguration(configuration) {
+  const { activeInstance, instances } = configuration;
+  return instances[activeInstance];
+}
+
+/**
+ * Generates a .env file string.
+ * @param {Record<string, string>} envConfiguration
+ * @returns {string}
+ */
+function buildEnvFile(envConfiguration) {
+  return (
+    Object.entries(envConfiguration)
+      .map(([key, value]) => [key, JSON.stringify(value)].join('='))
+      .join('\n') + '\n'
+  );
+}
+
+/**
+ * Parses the file at `filename` with dotenv.
+ * @param {string} filename
+ * @returns {Promise<import('dotenv').DotenvParseOutput>}
+ */
+async function readEnvFile(filename) {
+  const contents = await readFile(filename, 'utf-8');
+  const envConfig = parse(contents);
+  return envConfig;
+}
+
+/**
+ * Write the provided `config` to the .env file at `filename`, merging with any existing entries.
+ * @param {string} filename
+ * @param {Record<string, string>} config
+ */
+async function mergeEnvFiles(filename, config) {
+  const envFileExists = hasFile(filename);
+  const existingEnv = envFileExists ? await readEnvFile(filename) : {};
+  await writeFile(
+    join(process.cwd(), filename),
+    buildEnvFile({
+      ...existingEnv,
+      ...config,
+    }),
+  );
+}
+
+/**
+ * Installs the monorepo versions of the Clerk dependencies listed in the `package.json` file of the cwd.
+ * @returns {Promise<void>}
+ */
+async function linkDependencies() {
+  const { dependencies } = await getDependencies(join(process.cwd(), 'package.json'));
+  if (!dependencies) {
+    throw new Error('you have no dependencies');
+  }
+  const clerkPackages = await getClerkPackages();
+
+  const dependenciesToBeInstalled = Object.keys(dependencies)
+    .filter(dep => dep in clerkPackages)
+    .map(clerkDep => clerkPackages[clerkDep]);
+
+  const args = ['install', '--no-audit', '--no-fund', ...dependenciesToBeInstalled];
+
+  return new Promise((resolve, reject) => {
+    const child = spawn('npm', args, {
+      stdio: 'inherit',
+      env: {
+        ...process.env,
+        ADBLOCK: '1',
+        DISABLE_OPENCOLLECTIVE: '1',
+      },
+    });
+
+    child.on('close', code => {
+      if (code !== 0) {
+        reject();
+        return;
+      }
+      resolve();
+    });
+  });
+}
+
+/**
+ * Installs the monorepo-versions of Clerk dependencies listed in the `package.json` file of the current working
+ * directory, and performs framework configuration tasks necessary for local development with the monorepo packages.
+ * @param {object} args
+ * @param {boolean | undefined} args.js If `false`, do not customize the clerkJSUrl.
+ */
+export async function setup({ js = true }) {
+  console.log('Installing monorepo versions of Clerk packages from package.json...');
+  await linkDependencies();
```

Comment:
I've added a `--skip-install` option to allow for this! The idea is that `clerk-dev setup` will intelligently choose the right installation mechanism depending on framework. Next.js and Vite seem to work fine with `npm install`, so that's the default at the moment, but once we add support for other frameworks you'll see this command branch into other tools like `secco`.

But, in the event you want the framework configuration but _not_ the dependency configuration, you can now use `clerk-dev setup --skip-install`.